src/sfall_opcodes.cc

```diff
@@ -328,6 +329,72 @@ static void op_abs(Program* program)
     }
 }
 
+// sin
+static void op_sin(Program* program)
+{
+    ProgramValue programValue = programStackPopValue(program);
+    programStackPushFloat(program, sinf(programValue.asFloat()));
+}
+
+// cos
+static void op_cos(Program* program)
+{
+    ProgramValue programValue = programStackPopValue(program);
+    programStackPushFloat(program, cosf(programValue.asFloat()));
+}
+
+// tan
+static void op_tan(Program* program)
+{
+    ProgramValue programValue = programStackPopValue(program);
+    programStackPushFloat(program, tanf(programValue.asFloat()));
+}
+
+// arctan
+static void op_arctan(Program* program)
+{
+    ProgramValue xValue = programStackPopValue(program);
+    ProgramValue yValue = programStackPopValue(program);
+    programStackPushFloat(program, atan2f(yValue.asFloat(), xValue.asFloat()));
+}
+
+// pow (^)
+static void op_power(Program* program)
+{
+    ProgramValue expValue = programStackPopValue(program);
+    ProgramValue baseValue = programStackPopValue(program);
+
+    // CE: Implementation is slightly different, check.
+    float result = powf(baseValue.asFloat(), expValue.asFloat());
+
+    if (baseValue.isInt() && expValue.isInt()) {
+        programStackPushInteger(program, static_cast<int>(result));
```

Comment:
This matches sfall but seems like a bug to me.  2^-1 should == 0.5, but this will return 0.

cc @novarain @phobos2077

Reply:
https://github.com/sfall-team/sfall/blob/master/artifacts/scripting/sfall%20function%20notes.md#-operator-exponentiation
> If both arguments are integers, the result will be an integer.

Reply:
K, I can match that behaviour